src/traces/traces.hpp

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include <vector>
+
+#include "../common/logger.hpp"
+#include "../common/status.hpp"
+#include "../common/types.hpp"
+
+enum class TraceOperator {
+    insert,
+    search,
+    remove,
+};
+
+struct Trace {
+    TraceOperator op;
+    KeyType key;
+    ValueType value;
+
+    void
+    print() const;
+
+    inline bool
+    operator==(const Trace& rhs)
+    {
+        return this->op == rhs.op && this->key == rhs.key && this->value == rhs.value;
+    }
```

Comment:
```suggestion
    constexpr bool
    operator==(const Trace& rhs) const = default;
```
The compiler will do this exact implementation if you ask it to generate a default one.